src/main/java/org/folio/clients/impl/SourceRecordStorageClient.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.clients.impl;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.URI;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.util.EntityUtils;
+import org.folio.clients.SynchronousOkapiClient;
+import org.folio.util.OkapiConnectionParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * The client that synchronously communicates with the Source Record Storage(SRS).
+ * <p>
+ * Retrieves collection of underlying SRS records as a source of truth.
+ * Retrieves collection of Inventory records that do not have underlying SRS records
+ *
+ * @link https://github.com/folio-org/mod-source-record-storage
+ */
+@Component
+public class SourceRecordStorageClient extends SynchronousOkapiClient {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  public static final String GET_RECORD_PATTERN = "%s/source-storage/formattedRecords/%s?identifier=INSTANCE";
+
+  public SourceRecordStorageClient(OkapiConnectionParams okapiConnectionParams) {
+    super(okapiConnectionParams);
+  }
+
+  @Override
+  protected void prepareRequest(HttpRequestBase requestBase, String id) {
+    requestBase.setURI(URI.create(String.format(GET_RECORD_PATTERN,
```

Comment:
Also like discussed on slack already, try if we can use API similar to 
`curl --location --request GET 'https://folio-testing-okapi.aws.indexdata.com:443/inventory/instances?query=id=%22e6bc03c6-c137-4221-b679-a7c5c31f986c%22%20or%20id=%225bf370e0-8cca-4d9c-82e4-5170ab2a0a39%22' \
--header 'x-okapi-token: {{okapitoken}}'`
It will help in reducing total number of calls

Reply:
Last time I checked you can use a maximum of 15 ids per request with or

Reply:
The code has been chaged to use the URI stated above.